downloader/src/main/kotlin/vcs/GitWorkingTree.kt

```diff
@@ -21,68 +21,85 @@ package com.here.ort.downloader.vcs
 
 import com.here.ort.downloader.WorkingTree
 import com.here.ort.model.VcsInfo
-import com.here.ort.utils.ProcessCapture
 
 import java.io.File
 
-private const val GIT_STATUS_BRANCH_UPSTREAM_PREFIX = "# branch.upstream "
+import org.eclipse.jgit.api.LsRemoteCommand
+import org.eclipse.jgit.lib.BranchConfig
+import org.eclipse.jgit.lib.Constants
+import org.eclipse.jgit.lib.Repository
+import org.eclipse.jgit.storage.file.FileRepositoryBuilder
+import org.eclipse.jgit.submodule.SubmoduleWalk
 
 open class GitWorkingTree(workingDir: File, private val gitBase: GitBase) : WorkingTree(workingDir, gitBase.type) {
-    override fun isValid(): Boolean {
-        if (!workingDir.isDirectory) {
-            return false
+    private val repo = FileRepositoryBuilder().readEnvironment().findGitDir(workingDir).runCatching {
+        build()
+    }.getOrNull()
```

Comment:
I'm not sure in what situations this would file, e.g. if it is sometimes expected to fail, but do we really want to silently ignore this? If so, maybe explain that in a comment.

Reply:
We don't really silently ignore an error here, we just (implicitly) catch it later when `repo` is being used (but is `null`).

Reply:
I just thought it could be useful to print the exception to a debug log.